x/datapool/keeper/pool_distribute.go

```diff
@@ -0,0 +1,177 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/medibloc/panacea-core/v2/types/assets"
+	"github.com/medibloc/panacea-core/v2/x/datapool/types"
+)
+
+func (k Keeper) DistributeRevenuePools(ctx sdk.Context) error {
+	revenuePools, err := k.getDistributeRevenuePools(ctx)
+	if err != nil {
+		return err
+	}
+
+	if revenuePools.IsEmpty() {
+		return nil
+	}
+
+	// This lastIndex is to execute the following iteration in the next block if it does not end in one block.
+	// But We're currently processing it all in one block. :)
+	lastIndex := 0
+	// TODO We need to think about how to deal with the pool where distribution has failed.
+	for i, poolID := range revenuePools.PoolIds {
+		lastIndex = i
+		// search current pool info
+		pool, err := k.GetPool(ctx, poolID)
+		if err != nil {
+			return err
+		}
+
+		// calculate the revenue to be sent to each seller(totalSalesBalance / totalShareTokenAmount)
+		eachDistributionAmount, err := k.getEachDistributionAmount(ctx, pool)
+		if err != nil {
+			return err
+		}
+
+		// send deposit to curator
+		err = k.sendDepositToCurator(ctx, pool)
+		if err != nil {
+			return err
+		}
+
+		// distribute of revenue to each seller
+		k.bankKeeper.IterateAllBalances(ctx, func(addr sdk.AccAddress, balance sdk.Coin) bool {
+			if balance.Denom != types.GetDenomOfShareToken(poolID) {
+				return false
+			}
+			// Seller must receive a distribution equal to the shareTokens it owns.
+			distributeCoins := sdk.NewCoins(
+				sdk.NewCoin(assets.MicroMedDenom, eachDistributionAmount.Mul(balance.Amount)),
+			)
+
+			poolAddress, err := sdk.AccAddressFromBech32(pool.GetPoolAddress())
+			// stop distribute when an error occurs
+			if err != nil {
+				return true
+			}
+
+			// excluding moduleAccount
+			if k.accountKeeper.GetModuleAddress(types.ModuleName).Equals(addr) {
```

Comment:
You can use `types.GetModuleAddress()` which may returns the same address.

Reply:
Oh good. Thank you.
I'll changed it

Reply:
Completed!
https://github.com/medibloc/panacea-core/pull/311/commits/9ae0249839142138a4922cd0c4ad10a992a44b5b